File: java/src/main/java/io/darkstrata/credentialcheck/exception/ValidationException.java
package io.darkstrata.credentialcheck.exception;

/**
 * Exception thrown when input validation fails.
 */
public class ValidationException extends DarkStrataException {

    private final String field;

    public ValidationException(String message) {
        this(message, null);
    }

    public ValidationException(String message, String field) {
        super(message, ErrorCode.VALIDATION_ERROR, null, false);
        this.field = field;
    }

    /**
     * Get the field that failed validation, if applicable.
     */
    public String getField() {
        return field;
    }
}

File: sdks/csharp/src/DarkStrata.CredentialCheck/Exceptions/ValidationException.cs
namespace DarkStrata.CredentialCheck;

/// <summary>
/// Exception thrown when input validation fails.
/// </summary>
public sealed class ValidationException : DarkStrataException
{
    /// <summary>
    /// The field that failed validation.
    /// </summary>
    public string? Field { get; }

    /// <summary>
    /// Creates a new ValidationException.
    /// </summary>
    /// <param name="message">The error message.</param>
    /// <param name="field">The field that failed validation.</param>
    public ValidationException(string message, string? field = null)
        : base(message, ErrorCode.ValidationError, isRetryable: false)
    {
        Field = field;
    }
}
